File: front-end/src/app/core/interfaces/iauth.service.ts
import { Observable } from 'rxjs';
import { LoginRequest, LoginResponse, SetupPasswordRequest } from '../models/auth.model';
import { Utilisateur } from '../models/utilisateur.model';

export interface IAuthService {
  login(request: LoginRequest): Observable<LoginResponse>;

  logout(): Observable<void>;

  me(): Observable<Utilisateur>;

  setupPassword(request: SetupPasswordRequest): Observable<void>;
}

File: backend/src/main/java/com/example/backend/service/interfaces/IAuthService.java
package com.example.backend.service.interfaces;

import com.example.backend.dto.auth.AuthResponse;
import com.example.backend.dto.auth.LoginRequest;
import com.example.backend.dto.utilisateur.UtilisateurResponse;
import jakarta.servlet.http.HttpServletResponse;

public interface IAuthService {
    AuthResponse login(LoginRequest request, HttpServletResponse response);

    void logout(HttpServletResponse response);

    UtilisateurResponse getCurrentUser();
}
